tiger_agent/migrations/idempotent/002-admin.sql

```diff
@@ -0,0 +1,88 @@
+--002-admin.sql
+
+-----------------------------------------------------------------------
+-- function to extract user ID from Slack event in format <@U06S8H0V94P|nathan>
+create or replace function agent.extract_user_id_from_event(_event jsonb) returns text
+as $func$
+    select (regexp_match(_event->>'text', '<@([A-Z0-9]+)\|[^>]+>'))[1]
+$func$ language sql immutable security invoker
+;
+
+-----------------------------------------------------------------------
+-- agent.admin_users
+create table if not exists agent.admin_users
+( event_ts timestamptz not null default now()
+, user_id text primary key
+, event jsonb
+);
+
+-----------------------------------------------------------------------
+-- slack.insert_admin_user
+create or replace function agent.insert_admin_user(_event jsonb) returns void
+as $func$
+    insert into agent.admin_users (user_id, event)
+    select agent.extract_user_id_from_event(_event), _event
+    where agent.extract_user_id_from_event(_event) is not null
+    on conflict (user_id) do nothing
+$func$ language sql volatile security invoker
+;
+
+-----------------------------------------------------------------------
+-- agent.ignored_users
+create table if not exists agent.ignored_users
+( id serial primary key
+, event_ts timestamptz not null default now()
+, user_id text not null
+, deleted timestamptz null
```

Comment:
Please, please don't use soft deletes. If you want to keep a log, move the deleted records to an "audit/log/dead" table.

Reply:
The ergonomics on soft-delete tables are horrible. IMO it's always an anti-pattern.